### PR TITLE
DDF-04301 Add support for preQueryPlugin to be async

### PR DESCRIPTION
**Pull request by willwill96**
#### What does this PR do?
Adds support for preQueryPlugin to be an async function. This will give more flexibility to what the plugin can do

#### Who is reviewing it? 

@gjvera 
@mojogitoverhere 
@rymach 

#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@djblue

#### How should this be tested?

Spot test search functionality: 
  - try creating (from both the search bar and a blank workspace), editing, running and saving queries and look for regressions
#### What are the relevant tickets?
For GH Issues:
Fixes: #4301 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by coyotesqrl** [comment]
I've left some questions on the Issue, #4301

File: ui/packages/catalog-ui-search/src/main/webapp/js/QueryPolling.js [hunk 1]
```diff
@@ -175,29 +175,31 @@ startSearch = function(originalQuery, timeRange, queryToRun) {
...
+        var metacardIds = queryToRun
+          .get('result')
+          .get('results')
+          .map(function(result) {
+            return result
+              .get('metacard')
+              .get('properties')
+              .get('id')
+          })
+        var when = Date.now()
```
Comment (gjvera) [inline ui/packages/catalog-ui-search/src/main/webapp/js/QueryPolling.js]: ✏️ const

File: ui/packages/catalog-ui-search/src/main/webapp/js/QueryPolling.js [hunk 2]
```diff
@@ -175,29 +175,31 @@ startSearch = function(originalQuery, timeRange, queryToRun) {
...
-          workspaceId: originalQuery.collection.parents[0].id,
-          when: when,
-          metacardIds: metacardIds,
-        })
+  queryToRun.startSearch(undefined, searches => {
+    $.whenAll.apply(this, searches).always(function() {
+      if (pollingQueries[originalQuery.id]) {
+        checkForFailures(arguments, originalQuery, timeRange)
+        queryToRun.get('result').mergeNewResults()
+        var metacardIds = queryToRun
```
Comment (gjvera) [inline ui/packages/catalog-ui-search/src/main/webapp/js/QueryPolling.js]: ✏️ const

**Comment by mojogitoverhere** [comment]
build now

**Comment by gjvera** [comment]
🎉 hero successful 🎉 
There were no regressions in created a basic, advanced, or text search.
as well as no regressions in creating a search from the workspaces page.

LGTM!!

**Comment by bdeining** [comment]
@willwill96 need to update the commit message to have the ticket number

**Comment by willwill96** [comment]
@bdeining done

**Comment by gjvera** [comment]
build now

**Pull request merged**